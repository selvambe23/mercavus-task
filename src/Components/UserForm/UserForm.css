.userAddform{
    display: flex;
    align-items: flex-start;
    flex-wrap: nowrap;
    flex-direction: row;
}
.userAddform button {
    margin-bottom: 0.5em;
    background: #fafafa;
    color: #333;
    font-size: 1em;
    font-weight: bold;
    padding: 14px 25px;
    border-radius: 10px;
    text-decoration: none;
    display: inline-block;
  }
  .nameBox {
    padding: 15px 25px;
    border: 2px solid #aaa;
    border-radius: 10px;
    margin-right: 20px;
    font-size: 18px;
  }