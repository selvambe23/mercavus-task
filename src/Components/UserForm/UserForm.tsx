import React, {useState} from "react";
import "./UserForm.css";

const UserForm:  React.FC<{ userAddFunc: any }> = ({ userAddFunc }) => {
    const [name, setName] = useState<any>('');

    const UserForm = <div className="userAddform">
                        <input
                        type="text"
                        value={name}
                        onChange={(e) => setName(e.target.value)}
                        className="nameBox" />
                        <button onClick={(e) => userAddFunc(e, name)}>Add</button>
                </div>
    return UserForm;
};

export default UserForm;