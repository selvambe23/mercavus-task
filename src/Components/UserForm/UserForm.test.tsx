import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from "react-test-renderer";
import UserForm from './UserForm';

describe('UserForm Component', () => {
  it('renders snapshots, too', () => {
    const userAddFunc = jest.fn();
    const wrapper = shallow(<UserForm userAddFunc={userAddFunc} />);
    expect(wrapper.find('.userAddform')).toBeDefined();

    const tree = renderer
      .create(<UserForm userAddFunc={userAddFunc} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
});