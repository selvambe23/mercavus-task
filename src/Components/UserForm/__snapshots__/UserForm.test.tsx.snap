// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm Component renders snapshots, too 1`] = `
<div
  className="userAddform"
>
  <input
    className="nameBox"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
